### PR TITLE
Changelog 24.9.0

### PR DESCRIPTION
**Pull request by ForgottenProgramme**
### Description

✂️ snip snip ✂️ the making of a new release.

Xref #14232

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,184 @@
...
+* Mark `conda.gateways.disk.delete.move_to_trash` as pending deprecated. Use `conda.gateways.disk.delete.rm_rf` instead. (#14094)
+* Mark `conda.gateways.disk.delete.move_path_to_trash` as pending deprecated. Use `conda.gateways.disk.delete.rm_rf` instead. (#14094)
+* Mark `conda.exports.move_to_trash` as pending deprecation. Use `conda.gateways.disk.delete.rm_rf` instead. (#14118)
+Mark `conda.activate.ensure_binary` as pending deprecation. Use `conda.common.compat.ensure_binary` instead. (#14144)
+Mark `conda.activate.ensure_fs_path_encoding` as pending deprecation. (#14144)
+Mark `conda.common.compat.six_with_metaclass` as pending deprecation. Use class' `metaclass=` keyword argument instead. (#14144)
+Mark `conda.common.compat.open` as pending deprecation. Use `conda.common.compat.open_utf8` instead. (#14144, #14169)
+Mark `conda.common.compat.ensure_unicode` as pending deprecation. (#14144)
+Mark `conda.common.compat.ensure_fs_path_encoding` as pending deprecation. (#14144)
+Mark `conda.common.compat.FILESYSTEM_ENCODING` as pending deprecation. (#14144, #14169)
```
Comment (kenodegard) [inline CHANGELOG.md:111]:
```suggestion
* Mark `conda.activate.ensure_binary` as pending deprecation. Use `conda.common.compat.ensure_binary` instead. (#14144)
* Mark `conda.activate.ensure_fs_path_encoding` as pending deprecation. (#14144)
* Mark `conda.common.compat.six_with_metaclass` as pending deprecation. Use class' `metaclass=` keyword argument instead. (#14144)
* Mark `conda.common.compat.open` as pending deprecation. Use `conda.common.compat.open_utf8` instead. (#14144, #14169)
* Mark `conda.common.compat.ensure_unicode` as pending deprecation. (#14144)
* Mark `conda.common.compat.ensure_fs_path_encoding` as pending deprecation. (#14144)
* Mark `conda.common.compat.FILESYSTEM_ENCODING` as pending deprecation. (#14144, #14169)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,184 @@
 [//]: # (current developments)
 
+## 24.9.0 (2024-09-26)
+
+### Enhancements
+
+* Add `conda.core.index.Index` as a faster drop-in replacement of the realized dictionary index. (#13880, #14267)
+
+  Note: This removes the implicit initialization of the logging from `get_index`. If you relied on that, you should probably call `conda.gateways.logging.initialize_logging` explicitly.
```
Comment (kenodegard) [inline CHANGELOG.md:9]: The newline here forces the entire list to have extra newlines. Let's rephrase this into one line:

```suggestion
* Add `conda.core.index.Index` as a faster drop-in replacement of the realized dictionary index. Note: The loggers are no longer implicitly initialized when fetching the index, instead explicitly call `conda.gateways.logging.initialize_logging`. (#13880, #14267)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,184 @@
...
+* Deprecate `conda.env.specs.binstar` module. (#14158 via #14160)
+* Deprecate `conda env [create|update] REMOTE_DEFINITION`. Use `conda env [create|update] --file=URL` instead. (#14158 via #14160)
+* Mark `conda.testing.integration.BIN_DIRECTORY` as pending deprecation. Use `conda.common.path.BIN_DIRECTORY` instead. (#14188)
+* Mark `conda.common.path.get_bin_directory_short_path()` as pending deprecation. Use `conda.common.path.BIN_DIRECTORY` instead. (#14188)
+* Mark `conda.common.path.is_private_env_name` as pending deprecation. (#14189)
+* Mark `conda.common.path.is_private_env_path` as pending deprecation. (#14189)
+* Mark `conda shell.SHELL commands` as pending deprecation. Use `conda commands` instead. (#14215)
+* The `defaults` multichannel will stop being the (implicit) default value for `channels`. Users
+  relying on this behaviour are encouraged to run `conda config --add channels defaults`.
+  This is now pending deprecation, and will be fully deprecated in 25.3. (#14178 via #14227)
```
Comment (kenodegard) [inline CHANGELOG.md:133]:
```suggestion
* The `defaults` multichannel will stop being the (implicit) default value for `channels`. Users relying on this behavior are encouraged to run `conda config --add channels defaults`. This is pending deprecation, and will be fully deprecated in 25.3. (#14178 via #14227)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,184 @@
...
+* Running `conda config --add/--append channels ...` will warn when `defaults` is being
+  added implicitly. By 25.3, this behaviour will be removed and users should run
+  `conda config --add/--append channels defaults` explicitly if needed. Conda
+  distribution installers like miniforge or miniconda will pre-
+  configure conda channels during installation. (#12356 via #14227)
```
Comment (kenodegard) [inline CHANGELOG.md:138]:
```suggestion
* `conda config --add/--append channels ...` will warn when `defaults` is implicitly added. In `conda 25.3`, this behavior will be removed and users should run `conda config --add/--append channels defaults` explicitly if needed. Conda distribution installers like miniforge or miniconda will pre-configure conda channels during installation. (#12356 via #14227)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,184 @@
...
+* Without an explicit `channels` configuration (via `condarc` files, environment
+  variables, or CLI flags), conda will warn about using `defaults` implicitly.
+  In 25.3, an empty list will be used. (#14227)
```
Comment (kenodegard) [inline CHANGELOG.md:141]:
```suggestion
* Without an explicit `channels` configuration (via `condarc` files, environment variables, or CLI flags), conda will warn about using `defaults` implicitly. In 25.3, an empty list will be used. (#14227)
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,184 @@
...
+* Add `conda.testing.fixtures.session_conda_cli`. Use this to invoke conda commands within module, package, and session scoped fixtures. (#14243)
+* Add `conda.testing.fixtures.session_tmp_env`. Use this to create a conda environment within module, package, and session scoped fixtures. (#14243)
+
+### Bug fixes
+
+* Prevent directories which contain conda environments from being specified as an environment prefix when creating new envs; this provides guardrails against the accidental deletion of environments via commands such as `conda rename`. (#13955)
+* Enable `conda doctor` to check whether the environment exists or not before trying to generate a health report. (#14042 via #14112)
+* Fix `conda.common.configuration.ObjectParameter`'s ability to appropriately handle defaults. (#14128 via #14149)
+* `conda remove [env] --all` command no longer deletes empty parent directories for environments that are removed. (#14173)
+* Raise error when attempting to remove non-existent environment (#14174 via #14199)
```
Comment (beeankha) [inline CHANGELOG.md:28]:
```suggestion
* Raise an error when attempting to remove non-existent environment. (#14174 via #14199)
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,174 @@
...
+* Mark `conda.core.subdir_data.make_feature_record` as pending deprecation. Use `conda.models.records.PackageRecord.feature` instead. (#13880)
+* Mark `conda.plugins.manager.get_virtual_packages` as pending deprecation. Use `conda.plugins.manager.get_virtual_package_records` instead. (#13880)
+* Mark `conda.misc.explicit(index_args)` as pending deprecation. (#14267)
+* Mark `conda.cli.main_rename.validate_destination` as pending deprecation. Use `conda.cli.install.validate_new_prefix` instead. (#13955)
+* Mark `conda.cli.main_env_remove.execute` as pending deprecation. Use `conda.cli.main_remove.execute` instead. (#13977)
+* Mark `conda.activate.path_identity` as pending deprecation. Use `conda.common.path.path_identity` instead. (#14068)
+* Mark `conda.utils.path_identity` as pending deprecation. Use `conda.common.path.path_identity` instead. (#14068)
+* Mark `conda.models.leased_path_entry` as pending deprecation. (#14077)
+* Postpone `conda.base.context.Context.conda_exe` to `conda 25.3`. (#14077)
+* Postpone `conda.cli.python_api` to `conda 25.9`. (#14077)
```
Comment (beeankha) [inline CHANGELOG.md:50]:
```suggestion
* Postpone removal of `conda.base.context.Context.conda_exe` to `conda 25.3`. (#14077)
* Postpone removal of `conda.cli.python_api` to `conda 25.9`. (#14077)
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,174 @@
...
+* Mark `conda.testing.tmp_pkgs_dir` as pending deprecation. Use `conda.testing.fixtures.tmp_pkgs_dir` instead. (#14243)
+* Mark `conda.testing.tmp_envs_dir` as pending deprecation. Use `conda.testing.fixtures.tmp_envs_dir` instead. (#14243)
+
+### Docs
+
+* Document `--platform` flag for `conda [env] create`. (#14100 via #14181)
+
+### Other
+
+* Update xonsh support to accomodate deprecated import path. (#14047)
```
Comment (beeankha) [inline CHANGELOG.md:151]:
```suggestion
* Update `xonsh` support to accomodate deprecated import path. (#14047)
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,174 @@
...
+* Mark `conda.core.index.fetch_index` as pending deprecation. Use `conda.core.index.Index` instead. (#13880)
+* Mark `conda.core.index._supplement_index_with_prefix` as pending deprecation. Use `conda.core.index.Index.reload` instead. (#13880)
+* Mark `conda.core.index._supplement_index_with_cache` as pending deprecation. Use `conda.core.index.Index.reload` instead. (#13880)
+* Mark `conda.core.index._supplement_index_with_features` as pending deprecation. Use `conda.core.index.Index.reload` instead. (#13880)
+* Mark `conda.core.index._supplement_index_with_system` as pending deprecation. Use `conda.core.index.Index.reload` instead. (#13880)
+* Mark `conda.core.index._make_virtual_package` as pending deprecation. Use `conda.models.records.PackageRecord.virtual_package` instead. (#13880)
+* Mark `conda.core.subdir_data.make_feature_record` as pending deprecation. Use `conda.models.records.PackageRecord.feature` instead. (#13880)
+* Mark `conda.plugins.manager.get_virtual_packages` as pending deprecation. Use `conda.plugins.manager.get_virtual_package_records` instead. (#13880)
```
Comment (beeankha) [inline CHANGELOG.md:42]:
```suggestion
* Mark `conda.plugins.manager.CondaPluginManager.get_virtual_packages` as pending deprecation. Use `conda.plugins.manager.CondaPluginManager.get_virtual_package_records` instead. (#13880)
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,174 @@
...
+* Add `conda commands` subcommand. (#14215)
+* Add `conda.testing.fixtures.session_capsys`. Use this to capture stdout and stderr within module, package, and session scoped fixtures. (#14243)
+* Add `conda.testing.fixtures.session_conda_cli`. Use this to invoke conda commands within module, package, and session scoped fixtures. (#14243)
+* Add `conda.testing.fixtures.session_tmp_env`. Use this to create a conda environment within module, package, and session scoped fixtures. (#14243)
+
+### Bug fixes
+
+* Prevent directories which contain conda environments from being specified as an environment prefix when creating new envs; this provides guardrails against the accidental deletion of environments via commands such as `conda rename`. (#13955)
+* Enable `conda doctor` to check whether the environment exists or not before trying to generate a health report. (#14042 via #14112)
+* Fix `conda.common.configuration.ObjectParameter`'s ability to appropriately handle defaults. (#14128 via #14149)
```
Comment (kenodegard) [inline CHANGELOG.md:24]: This refers to the wrong issue, we want https://github.com/conda/conda/issues/14148 not https://github.com/conda/conda/issues/14128:

```suggestion
* Fix `conda.common.configuration.ObjectParameter`'s ability to appropriately handle defaults. (#14148 via #14149)
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,174 @@
 [//]: # (current developments)
 
+## 24.9.0 (2024-09-26)
+
+### Enhancements
+
+* Add `conda.core.index.Index` as a faster drop-in replacement of the realized dictionary index. Note: The loggers are no longer implicitly initialized when fetching the index, instead explicitly call `conda.gateways.logging.initialize_logging`. (#13880, #14267)
```
Comment (kathatherine) [inline CHANGELOG.md:7]:
```suggestion
* Add `conda.core.index.Index` as a faster drop-in replacement of the realized dictionary index. Note: The loggers are no longer implicitly initialized when fetching the index. Instead, you must explicitly call `conda.gateways.logging.initialize_logging`. (#13880, #14267)
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,174 @@
...
+* Fix Windows to Unix path conversion to handle UNC mounts and root paths. (#14157)
+* Add a new `denylist_channels` config option and `CONDA_DENYLIST_CHANNELS` environment variable to explicitly deny using specific channels globally, which is complementing the already existing `allowlist_channels` config option. (#14176 via #14196)
+* Add `conda commands` subcommand. (#14215)
+* Add `conda.testing.fixtures.session_capsys`. Use this to capture stdout and stderr within module, package, and session scoped fixtures. (#14243)
+* Add `conda.testing.fixtures.session_conda_cli`. Use this to invoke conda commands within module, package, and session scoped fixtures. (#14243)
+* Add `conda.testing.fixtures.session_tmp_env`. Use this to create a conda environment within module, package, and session scoped fixtures. (#14243)
+
+### Bug fixes
+
+* Prevent directories which contain conda environments from being specified as an environment prefix when creating new envs; this provides guardrails against the accidental deletion of environments via commands such as `conda rename`. (#13955)
```
Comment (kathatherine) [inline CHANGELOG.md:22]:
```suggestion
* Prevent directories that contain conda environments from being specified as an environment prefix when creating new envs; this provides guardrails against the accidental deletion of environments via commands such as `conda rename`. (#13955)
```

**Review by kathatherine — COMMENTED** [review]
Just a few minor things on my end

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,5 +1,174 @@
...
+* Remove `conda.core.subdir_data.get_cache_control_max_age`. Use `conda.gateways.repodata.get_cache_control_max_age` instead. (#14077)
+* Remove `conda.exports.fetch_index`. Use `conda.core.index.fetch_index` instead. (#14077)
+* Remove `conda.exports.IndexRecord`. Use `conda.models.records.PackageRecord` instead. (#14077)
+* Remove `conda.gateways.anaconda_client.EnvAppDirs`. Use `platformdirs` instead. (#14077)
+* Remove `conda.gateways.connection.adapters.ftp.FTPAdapter.stor`. (#14077)
+* Remove `conda.gateways.connection.adapters.ftp.parse_multipart_files` (#14077)
+* Remove `conda.gateways.logging.set_verbosity`. Use `conda.gateways.logging.set_log_level` instead. (#14077)
+* Remove `conda.gateways.logging.VERBOSITY_LEVELS`. (#14077)
+* Remove `conda.models.dist.IndexRecord`. Use `conda.models.records.PackageRecord` instead. (#14077)
+* Remove `conda.models.enum.LeasedPathType`. (#14077)
```
Comment (ForgottenProgramme) [inline CHANGELOG.md:89]: this should be "enums"

Comment (kenodegard) [reply CHANGELOG.md:89]:
```suggestion
* Remove `conda.models.enums.LeasedPathType`. (#14077)
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,5 +1,174 @@
 [//]: # (current developments)
 
+## 24.9.0 (2024-09-26)
+
+### Enhancements
```
Comment (jezdez) [inline CHANGELOG.md:5]:
```suggestion

### ✨ Special announcement ✨

This is an announcement about an important and positive **future** change in conda's functionality:

Following feedback from conda users around concerns about the pre-configuration of the conda code base to favor channels from Anaconda Inc, we've [started the process to deprecate](https://github.com/conda/conda/issues/14178) hardcoding Anaconda's channels as the default set of channels in the conda source code, which is a remnant of conda's incubation at the company. The conda ecosystem is able to facilitate the configuration of the channels through different means.

We will in the future rely on the fact that providers of conda distributions, such as like [miniforge](https://github.com/conda-forge/miniforge) or Anaconda (including miniconda) can pre-configure their used channels on their own, e.g. by running the neccary ``conda config --set channels`` command.

We're also going to continue to [work on improving channel management](https://github.com/conda/conda/issues/14217) in the forseeable future and would love to get your feedback about it.

### Enhancements
```

**Review by jezdez — DISMISSED** [review]
It would be splendid to get my suggested blurb added, but another set of eyes wouldn't hurt given the nuance and importance of the topic.

File: CHANGELOG.md (hunk 14 above)
Comment (beeankha) [reply CHANGELOG.md:5]: I would change the second to last paragraph to:

```
In the future, we will rely on the fact that providers of conda distributions, such as [miniforge](https://github.com/conda-forge/miniforge) or Anaconda (including miniconda) can pre-configure their used channels on their own, e.g. by running the necessary ``conda config --set channels`` command.
```

Comment (kenodegard) [reply CHANGELOG.md:5]: I'm going to commit the original suggested change as is, it'll be easier to edit further through code reviews

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,5 +1,185 @@
...
+## 24.9.0 (2024-09-26)
+
+
+### ✨ Special announcement ✨
+
+This is an announcement about an important and positive **future** change in conda's functionality:
+
+Following feedback from conda users around concerns about the pre-configuration of the conda code base to favor channels from Anaconda Inc, we've [started the process to deprecate](https://github.com/conda/conda/issues/14178) hardcoding Anaconda's channels as the default set of channels in the conda source code, which is a remnant of conda's incubation at the company. The conda ecosystem is able to facilitate the configuration of the channels through different means.
+
+We will in the future rely on the fact that providers of conda distributions, such as like [miniforge](https://github.com/conda-forge/miniforge) or Anaconda (including miniconda) can pre-configure their used channels on their own, e.g. by running the neccary ``conda config --set channels`` command.
```
Comment (kenodegard) [inline CHANGELOG.md:12]:
```suggestion
In the future, we will rely on providers of conda distributions, such as [miniforge](https://github.com/conda-forge/miniforge) or Anaconda (including miniconda), to pre-configure their preferred channels, e.g. by running the necessary ``conda config --set channels`` command.
```

Comment (beeankha) [inline CHANGELOG.md:12]:
```suggestion
In the future, we will rely on the fact that providers of conda distributions, such as [miniforge](https://github.com/conda-forge/miniforge) or Anaconda (including miniconda) can pre-configure their used channels on their own, e.g. by running the necessary ``conda config --set channels`` command.
```

Comment (beeankha) [reply CHANGELOG.md:12]: Oops, didn't see you had already made the above change suggestion.

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,5 +1,185 @@
 [//]: # (current developments)
 
+## 24.9.0 (2024-09-26)
+
+
+### ✨ Special announcement ✨
+
+This is an announcement about an important and positive **future** change in conda's functionality:
+
+Following feedback from conda users around concerns about the pre-configuration of the conda code base to favor channels from Anaconda Inc, we've [started the process to deprecate](https://github.com/conda/conda/issues/14178) hardcoding Anaconda's channels as the default set of channels in the conda source code, which is a remnant of conda's incubation at the company. The conda ecosystem is able to facilitate the configuration of the channels through different means.
```
Comment (kenodegard) [inline CHANGELOG.md:10]: I don't think this last sentence here is needed?

```suggestion
Following feedback from conda users concerns about the pre-configuration of the conda code base to favor channels from Anaconda Inc, we've [started the process to deprecate](https://github.com/conda/conda/issues/14178) hardcoding Anaconda's channels as the default set of channels in the conda source code, which is a remnant of conda's incubation at the company.
```

File: CHANGELOG.md [hunk 17]
```diff
@@ -1,5 +1,185 @@
...
+
+We will in the future rely on the fact that providers of conda distributions, such as like [miniforge](https://github.com/conda-forge/miniforge) or Anaconda (including miniconda) can pre-configure their used channels on their own, e.g. by running the neccary ``conda config --set channels`` command.
+
+We're also going to continue to [work on improving channel management](https://github.com/conda/conda/issues/14217) in the forseeable future and would love to get your feedback about it.
```
Comment (kenodegard) [inline CHANGELOG.md:14]:
```suggestion
We're also going to continue to [work on improving channel management](https://github.com/conda/conda/issues/14217) in the forseeable future and would love to get your feedback.
```

**Review by beeankha — APPROVED** [review]

**Review by kenodegard — APPROVED** [review]

**Comment by jezdez** [comment]
Thank you all for fixing my Denglish! 🤣

**Pull request merged**